Lib/http/server.py

```diff
@@ -686,10 +687,30 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
+        fs = os.fstat(f.fileno())
+            
         try:
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
```

Comment:
Self.headers should be (similar to) an email.message.Message object, so I think it should work fine